### PR TITLE
BUGFIX: EGL-62 apply Overlay for DateRangePicker

### PR DESCRIPTION
**Pull request by phong-nguyen-duy**
JIRA: EGL-62

---

Supported PR commands:

| Command                                                 | Description                                                |
| ------------------------------------------------------- | ---------------------------------------------------------- |
| `ok to test`                                            | Re-run standard checks                                     |
| `extended check sonar`                                  | SonarQube tests                                            |
| `extended test - backstop`                              | BackstopJS tests                                           |
| **E2E Cypress tests commands - TIGER**                  |                                                            |
| `extended test - tiger-cypress - isolated <testName>`   | Run isolated tests running against recorded Tiger backend. |
| `extended test - tiger-cypress - record <testName>`     | Create a new recording for isolated Tiger tests.           |
| `extended test - tiger-cypress - integrated <testName>` | Run integrated tests against live backend                  |
| **E2E Cypress tests commands - BEAR**                   |                                                            |
| `extended test - cypress - isolated <testName>`         | Run isolated tests running against recorded Bear backend.  |
| `extended test - cypress - record <testName>`           | Create a new recording for isolated Bear tests.            |
| `extended test - cypress - integrated <testName>`       | Run integrated tests against live backend                  |
| **Compatibility matrix test commands - TIGER Backend**  |                                                            |
| `extended test - matrix-test <AIO_version>`             | Run integrated tests against AIO versions.                 |

`<testName>` in cypress commands is used to filter specfiles. Example, to run record with BEAR backend

-   Against `dashboard.spec.ts` and `drilling.spec.ts`, execute command `extended test - cypress - record dashboard,drilling`
-   Against all specfiles, execute command `extended test - cypress - record` or `extended test - cypress - record *`

`<AIO_version>` in commands is used to start test with multiple AIO instances - each instance in triggered by one jenkins build

-   To run with `master` and `stable`, execute command `extended test - matrix-test master,stable` or `extended test - matrix-test latest`
-   To run with specific version,ex: `2.3.0` and `2.3.1`, execute command `extended test - matrix-test 2.3.0,2.3.1`
-   In case `<AIO_version>` is empty, read versions from file `compTigerVersions.txt` of this repo

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `extended test - backstop` passes
-   [ ] `extended test - tiger-cypress - record` to record new mapping files (Tiger BE)
-   [ ] `extended test - cypress - record` to record new mapping files (Bear BE)
-   [ ] `extended test - tiger-cypress - isolated` passes
-   [ ] `extended test - cypress - isolated` passes
-   [ ] `extended test - tiger-cypress - integrated` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/13814/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/15426/ : FAILURE in 8m 21s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/13815/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/15427/ : SUCCESS in 13m 29s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/13816/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/821/ : SUCCESS in 8m 44s

**Comment by phong-nguyen-duy** [comment]
extended test - cypress - isolated

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by phong-nguyen-duy** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/13828/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/15437/ : SUCCESS in 12m 02s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/13834/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/15445/ : SUCCESS in 12m 25s

**Comment by phong-nguyen-duy** [comment]
extended test - cypress - isolated

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/13835/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/827/ : SUCCESS in 12m 13s

File: libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateRangePicker.tsx [hunk 1]
```diff
@@ -191,10 +214,14 @@ class DateRangePickerComponent extends React.Component<DateRangePickerProps, IDa
...
                         {FromField}
-                        {isFromInputDatePickerOpen ? DatePicker : null}
+                        {isFromInputDatePickerOpen ? (
+                            <>{shouldOverlayDatePicker ? OverlayDatePicker : DatePicker}</>
+                        ) : null}
                         <label>{intl.formatMessage({ id: "filters.to" })}</label>
                         {ToField}
-                        {isToInputDatePickerOpen ? DatePicker : null}
+                        {isToInputDatePickerOpen ? (
+                            <>{shouldOverlayDatePicker ? OverlayDatePicker : DatePicker}</>
```
Comment (xMort) [inline libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateRangePicker.tsx:223]: I would extract this as a constant to not repeat this block twice, i.e., 
`const DatePickerComponent = shouldOverlayDatePicker ? OverlayDatePicker : DatePicker;`

Possibly the empty fragment will not be necessary.

File: libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateRangePicker.tsx [hunk 2]
```diff
@@ -150,6 +153,26 @@ class DateRangePickerComponent extends React.Component<DateRangePickerProps, IDa
             </div>
         );
 
+        const OverlayDatePicker = (
+            <Overlay
+                alignTo={`.gd-date-range-picker-${isTimeEnabled ? this.state.selectedInput : "from"}`}
+                alignPoints={[
```
Comment (xMort) [inline libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateRangePicker.tsx:159]: Please extract this object as constant outside of the main component.

File: libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateRangePicker.tsx [hunk 3]
```diff
@@ -150,6 +153,26 @@ class DateRangePickerComponent extends React.Component<DateRangePickerProps, IDa
             </div>
         );
 
+        const OverlayDatePicker = (
+            <Overlay
+                alignTo={`.gd-date-range-picker-${isTimeEnabled ? this.state.selectedInput : "from"}`}
```
Comment (xMort) [inline libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateRangePicker.tsx:158]: Just a question: Can't this be  only `.gd-date-range-picker-${this.state.selectedInput}`? I see it used as this in other places in this component.

File: libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateRangePicker.tsx (hunk 2 above)
Comment (phong-nguyen-duy) [reply libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateRangePicker.tsx]: Done

File: libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateRangePicker.tsx (hunk 1 above)
Comment (phong-nguyen-duy) [reply libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateRangePicker.tsx]: Done

**Review by xMort — APPROVED** [review]

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3796

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/2763/ : SUCCESS in 9m 13s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/15449/ : SUCCESS in 11m 53s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul-pipeline/311/ : SUCCESS in 12m 33s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/3920/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/645/ : FAILURE in 1m 35s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-isolated-zuul-pipeline/274/ : SUCCESS in 8m 42s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-compatibility-pipeline/149/ : SUCCESS in 4m 52s

**Pull request merged**